pkg/kv/cosmosdb/store.go

```diff
@@ -324,7 +330,7 @@ func (s *Store) Delete(ctx context.Context, partitionKey, key []byte) error {
 	_, err := s.containerClient.DeleteItem(ctx, pk, s.hashID(key), nil)
 	var respErr *azcore.ResponseError
 	if errors.As(err, &respErr) && respErr.StatusCode != http.StatusNotFound {
-		return err
+		return convertError(err)
```

Comment:
now as we call `convertError` and it will convert to our specific types - we can condition the result err and not convert the error twice to `azcore.ReponseError`

Reply:
Think you missed this one

Reply:
I mean you need to convert first - check if it is not found and return error if needed.
so we will not convert the error twice.
if 99% from the times we would expect not found - I'll keep this version